test/sqllogictest/transform/aggregation_nullability.slt

```diff
@@ -280,25 +280,24 @@ query T multiline
 EXPLAIN WITH(arity, join implementations) select t1.f1, sum(t2.f1), max(t2.f1) from t1 LEFT JOIN t2 ON t1.f1 = t2.f1 group by t1.f1 having max(t2.f1) >= 0;
 ----
 Explained Query:
-  Filter (#2 >= 0) // { arity: 3 }
-    Reduce group_by=[#0] aggregates=[sum(#0), max(#0)] // { arity: 3 }
-      Project (#0) // { arity: 1 }
-        Join on=(#0 = #1) type=differential // { arity: 2 }
-          implementation
-            %0:t1[#0]K » %1:t2[#0]K
-          ArrangeBy keys=[[#0]] // { arity: 1 }
-            Project (#0) // { arity: 1 }
-              Filter (#0) IS NOT NULL // { arity: 2 }
-                ReadStorage materialize.public.t1 // { arity: 2 }
-          ArrangeBy keys=[[#0]] // { arity: 1 }
-            Project (#0) // { arity: 1 }
-              Filter (#0) IS NOT NULL // { arity: 2 }
-                ReadStorage materialize.public.t2 // { arity: 2 }
+  Reduce group_by=[#0] aggregates=[sum(#0), max(#0)] // { arity: 3 }
+    Project (#0) // { arity: 1 }
+      Join on=(#0 = #1) type=differential // { arity: 2 }
+        implementation
+          %0:t1[#0]Kif » %1:t2[#0]Kiif
+        ArrangeBy keys=[[#0]] // { arity: 1 }
+          Project (#0) // { arity: 1 }
+            Filter (#0 >= 0) // { arity: 2 }
```

Comment:
This appears to be a mis-optimization, pushing the filter down (the `sum(#0)` still needs all inputs to be correct, even if the `max(#0)` does not). However, this change doesn't result from a change to a transform, so some other transform seems to be buggy.

Reply:
Ah, or maybe not, because it is a crazy `Reduce` where the expressions feeding the aggregate are just the group key. So the predicate `#2 >= 0` is actually a predicate on `#0` the group key, which is allowed to be pushed down. What a world!

Reply:
I guess what happened is that the improved analysis surfaced the `#0 = #2` constraint, which allowed the filter to be rewritten a group key.

Reply:
It does suggest we are missing a strength-reduction where a `Reduce` aggregate of the MIN/MAX/ANY/ALL flavors have an expression matching the key. In that case, we can convert it to a `Map`, and spare everyone the hierarchical aggregation.